### DISCUSSION TITLE
Setup my debian as a headset

### DISCUSSION THREAD

**Discussion by bilogic** [Arkq/bluez-alsa#740] [Q&A]
I started googling from here https://askubuntu.com/questions/845195/how-to-set-up-ubuntu-pc-as-bluetooth-headset-to-attend-calls

I'm not sure where to start as I'm unfamiliar with both bluez and alsa.
Any ideas?

Thank you.

**Comment by arkq** [comment]
> I'm not sure where to start as I'm unfamiliar with both bluez and alsa.

If you are not familiar with BlueZ and ALSA, I would recommend to stick with what Debian has by default - PipeWire. To be honest I'm not sure how to enable such feature in PipeWire, but from the desktop point of view using PipeWire will cause you less amount of troubles. If however you really need to use bluez-alsa, then compile it with:

```shell
./configure --enable-systemd --enable-upower --enable-msbc --enable-lc3-swb \
  --enable-rfcomm --with-systemdbluealsadargs='-S -p a2dp-sink -p hfp-hf -p hsp-hs'
make && sudo make install
```

Then enable BlueALSA system service:
```shell
sudo systemctl enable bluealsa
```

And reboot the system (just in case).

Now you should have BlueALSA system service running which will allow you to connect your phone over Bluetooth. For that use some GUI available in Debian, or `bluetoothctl` in case of headless setup. Refer to the wiki for more info: https://github.com/arkq/bluez-alsa/wiki/Bluetooth-Pairing-And-Connecting

After that you should have two SCO PCMs available for playback and capture. You can use them as follows:
```shell
aplay -D bluealsa:PROFILE=SCO ...
arecord -D bluealsa:PROFILE=SCO ...
```

For more detailed info regarding "bluealsa" ALSA device refer to bluez-alsa manual: https://github.com/arkq/bluez-alsa/blob/master/doc/bluealsa-plugins.7.rst

**Reply by bilogic** [reply]
Wow thanks man, i'm actually using CLI debian, proxmox to be exact. Let me give your tips a shot. Thank you.
